stubs/setuptools/pkg_resources/__init__.pyi

```diff
@@ -119,10 +125,15 @@ class EntryPoint:
     def resolve(self) -> Any: ...
 
 def find_distributions(path_item: str, only: bool = ...) -> Generator[Distribution, None, None]: ...
-def get_distribution(dist: Requirement | str | Distribution) -> Distribution: ...
+@overload
+def get_distribution(dist: _D) -> _D: ...
+@overload
+def get_distribution(dist: _PkgReqType) -> DistInfoDistribution: ...
 
 class Distribution(IResourceProvider, IMetadataProvider):
-    PKG_INFO: str
+    PKG_INFO: ClassVar[str]
+    # Initialized to None, but is not meant to be instantiated directly
+    egg_info: str
```

Comment:
I don't see this attribute at all (on setuptools 67.3.1).

Reply:
Found it, it's in `NullProvider`. If you provide no metadata to Distribution, it uses `EmptyProvider` which is a `NullProvider`

```
>>> type(pkg_resources.get_distribution("numpy")._provider)
<class 'pkg_resources.PathMetadata'>
>>> type(Distribution()._provider)
<class 'pkg_resources.EmptyProvider'>
>>> type(get_provider("numpy"))
<class 'pkg_resources.DefaultProvider'>
```
`PathMetadata`, `EmptyProvider` and `DefaultProvider` all have `NullProvider` in common